build/parser.rs

```diff
@@ -1036,24 +1037,24 @@ pub enum MavXmlElement {
     Extensions,
 }
 
-fn identify_element(s: &str) -> Option<MavXmlElement> {
+fn identify_element(s: &[u8]) -> Option<MavXmlElement> {
     use self::MavXmlElement::*;
     match s {
-        "version" => Some(Version),
-        "mavlink" => Some(Mavlink),
-        "dialect" => Some(Dialect),
-        "include" => Some(Include),
-        "enums" => Some(Enums),
-        "enum" => Some(Enum),
-        "entry" => Some(Entry),
-        "description" => Some(Description),
-        "param" => Some(Param),
-        "messages" => Some(Messages),
-        "message" => Some(Message),
-        "field" => Some(Field),
-        "deprecated" => Some(Deprecated),
-        "wip" => Some(Wip),
-        "extensions" => Some(Extensions),
+        b"version" => Some(Version),
```

Comment:
Isn't it better to transform the xml access to utf-8 over than using all our strings to bytes ?

Reply:
Do you mean to convert it to strings when we compare, or do you mean to use the Decoder of quick-xml/unescape_and_decode?

quick-xml does all access by default in bytes, i tried to only convert it to strings where it is needed.

Reply:
Ok, I think that we continue with the current version of the PR.